src/Type/Php/DateTimeDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types=1);
+
+namespace PHPStan\Type\Php;
+
+use DateTime;
+use DateTimeImmutable;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeUtils;
+
+class DateTimeDynamicReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return in_array($functionReflection->getName(), ['date_create_from_format', 'date_create_immutable_from_format'], true);
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		$defaultReturnType = ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getReturnType();
+
+		$format = $scope->getType($functionCall->args[0]->value);
```

Comment:
You should check that the call has at least 2 arguments so that this code doesn't crash with `date_create_from_format()`.

Reply:
What is the suggested handling if there aren't at least 2 parameters?

Reply:
Return `$defaultReturnType`.